python/pyarrow/tests/test_extension_type.py

```diff
@@ -342,12 +345,12 @@ def test_ext_type_as_py():
 
 def test_uuid_type_pickle(pickle_module):
```

Comment:
Can we test pickling of _both_ the canonical uuid type and the example uuid type? Since their implementations are quite different.

Reply:
Added identical test for canonical.